mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -0,0 +1,1572 @@
+                                (wcscmp(className.GetBSTR(), L"Shell_SecondaryTrayWnd") == 0) ||               // Windows 10 secondary taskbar
+                                (wcscmp(className.GetBSTR(), L"Taskbar.TaskbarFrameAutomationPeer") == 0) ||   // Windows 11 taskbar
+                                (wcscmp(className.GetBSTR(), L"Windows.UI.Input.InputSite.WindowClass") == 0); // Windows 11 21H2 taskbar
+    if (!taskbarClicked) {
+        return false;
+    }
+
+    if (taskbarAction == ACTION_SHOW_DESKTOP) {
+        ShowDesktop(hWnd);
+    } else if (taskbarAction == ACTION_ALT_TAB) {
+        SendAltTab();
+    } else if (taskbarAction == ACTION_TASK_MANAGER) {
+        OpenTaskManager(hWnd);
+    } else if (taskbarAction == ACTION_MUTE) {
+        ToggleVolMuted();
+    } else if (taskbarAction == ACTION_TASKBAR_AUTOHIDE) {
+        (void)ToggleTaskbarAutohide();
+    } else if (taskbarAction == ACTION_WIN_TAB) {
+        SendWinTab();
+    } else if (taskbarAction == ACTION_HIDE_ICONS) {
+        HideIcons();
+    } else {
+        Wh_Log(L"Error: Unknown taskbar action '%d'", taskbarAction);
+    }
+
+    return false;
+}
+
+////////////////////////////////////////////////////////////
+
+BOOL Wh_ModInit() {
+    Wh_Log(L">");
+
+    LoadSettings();
+
+    if (!WindowsVersionInit() || (g_taskbarVersion == UNKNOWN_TASKBAR)) {
+        Wh_Log(L"Unsupported Windows version, ModInit failed");
+        return FALSE;
+    }
+    // treat Windows 11 taskbar as on older windows
+    if ((g_taskbarVersion == WIN_11_TASKBAR) && g_settings.oldTaskbarOnWin11) {
+        g_taskbarVersion = WIN_10_TASKBAR;
+    }
+    Wh_Log(L"Using taskbar version: %d");
+
+    // init COM for UIAutomation and Volume control
+    if (FAILED(CoInitializeEx(NULL, COINIT_MULTITHREADED))) {
+        Wh_Log(L"COM initialization failed, ModInit failed");
+        return FALSE;
+    } else {
+        Wh_Log(L"COM initilized");
+    }
+
+    // init COM interface for UIAutomation
+    if (FAILED(CoCreateInstance(CLSID_CUIAutomation, NULL, CLSCTX_INPROC_SERVER, __uuidof(IUIAutomation), g_pUIAutomation.put_void())) || !g_pUIAutomation) {
+        Wh_Log(L"Failed to create UIAutomation COM instance, ModInit failed");
+        return FALSE;
+    } else {
+        Wh_Log(L"UIAutomation COM initilized");
+    }
+
+    // init COM interface for Volume control
+    const GUID XIID_IMMDeviceEnumerator = {0xA95664D2, 0x9614, 0x4F35, {0xA7, 0x46, 0xDE, 0x8D, 0xB6, 0x36, 0x17, 0xE6}};
+    const GUID XIID_MMDeviceEnumerator = {0xBCDE0395, 0xE52F, 0x467C, {0x8E, 0x3D, 0xC4, 0x57, 0x92, 0x91, 0x69, 0x2E}};
+    if (FAILED(CoCreateInstance(XIID_MMDeviceEnumerator, NULL, CLSCTX_INPROC_SERVER, XIID_IMMDeviceEnumerator, g_pDeviceEnumerator.put_void())) || !g_pDeviceEnumerator) {
+        Wh_Log(L"Failed to create DeviceEnumerator COM instance, ModInit failed");
+        return FALSE;
```

Comment:
`CoUninitialize` is missing in this case.

Reply:
I implemented a wrapper that takes of Uninitialization even if Wh_ModUninit is not called.